src/main/java/org/folio/circulation/resources/RequestByInstanceIdResource.java

```diff
@@ -269,47 +261,54 @@ private void createInstanceLevelRequests(RoutingContext routingContext) {
     return CompletableFuture.allOf(requestQueueFutures.toArray(new CompletableFuture[requestQueueFutures.size()]))
       .thenApply(x -> {
         Map<Item, RequestQueue> itemQueueSizeMap = new HashMap<>();
+        List<Item> failedQueuesItemList = new LinkedList<>(); //to preserve order it was found.
+
         for (Map.Entry<Item, CompletableFuture<Result<RequestQueue>>> entry : itemRequestQueueMap.entrySet()) {
           Result<RequestQueue> requestQueueResult = entry.getValue().join();
           if (requestQueueResult.succeeded()) {
             itemQueueSizeMap.put(entry.getKey(), requestQueueResult.value());
+          } else {
+            failedQueuesItemList.add(entry.getKey());
           }
         }
 
+        if (failedQueuesItemList.size() == requestQueueFutures.size()) {
+          log.error("Failed to find request queues for all items of instanceId {}",
+                        ((List<Item>) unsortedUnavailableItems).get(0).getInstanceId());
```

Comment:
Why does `InstanceRequestRelatedRecords.getUnsortedAvailableItems()` return a `Collection<Item>`? All other methods there and the member variables use `List<Item>`. I don't see any indication that we need to return a `Collection`. Changing to a `List<item>` would make the class consistent and we wouldn't have to cast here.

Reply:
Good catch. I was wondering about why the IDE automatically casted it for me as well...